lightning-background-processor/src/lib.rs

```diff
@@ -886,7 +886,7 @@ mod tests {
 		fn disconnect_socket(&mut self) {}
 	}
 
-	type ChannelManager = channelmanager::ChannelManager<Arc<ChainMonitor>, Arc<test_utils::TestBroadcaster>, Arc<KeysManager>, Arc<KeysManager>, Arc<KeysManager>, Arc<test_utils::TestFeeEstimator>, Arc<DefaultRouter<Arc<NetworkGraph<Arc<test_utils::TestLogger>>>, Arc<test_utils::TestLogger>, Arc<Mutex<TestScorer>>, (), TestScorer>>, Arc<test_utils::TestLogger>>;
+	type ChannelManager = channelmanager::ChannelManager<Arc<ChainMonitor>, Arc<test_utils::TestBroadcaster>, Arc<KeysManager>, Arc<KeysManager>, Arc<KeysManager>, Arc<test_utils::TestFeeEstimator>, Arc<DefaultRouter<Arc<NetworkGraph<Arc<test_utils::TestLogger>>>, Arc<test_utils::TestLogger>, MutexGuard<'static, TestScorer>, Arc<Mutex<TestScorer>>, (), TestScorer>>, Arc<test_utils::TestLogger>>;
```

Comment:
Hmm, I'm definitely not understanding lifetimes right here, seems like i still need a lifetime parameter despite the for <'a> D: 'a + DerefMut<..>

its most definitely not 'static thats just a placeholder for now

whats happening here is the usage of mutexguard here in the definition of a channel manager requires us to set the lifetimes for how long it is going to last, but from the implementation of lockable what we do want is that the mutex guard will last for as long as the lockable score's locked type, which is generic across all `'a`

compiler wants the lifetime upfront but we only want to add that lifetime later on